sphinx/addnodes.py

```diff
@@ -294,9 +294,19 @@ class desc_annotation(nodes.Part, nodes.Inline, nodes.FixedTextElement):
 # Leaf nodes for markup of text fragments
 #########################################
 
+# A list of classes inheriting :class:`desc_sig_element`. Each node class
+# is expected to be handled by the builder's translator class if the latter
+# does not inherit from SphinxTranslator.
+#
+# This list can be extended manually by third-party extensions or by subclassing
+# :class:`desc_sig_element` and using the meta-keyword `_sig_element=True`.
+SIG_ELEMENTS: list[type[desc_sig_element]] = []
```

Comment:
```suggestion
#: A list of classes inheriting :class:`desc_sig_element`. Each node class
#: is expected to be handled by the builder's translator class if the latter
#: does not inherit from SphinxTranslator.
#:
#: This list can be extended manually by third-party extensions or by subclassing
#: :class:`desc_sig_element` and using the meta-keyword `_sig_element=True`.
SIG_ELEMENTS: list[type[desc_sig_element]] = []
```

Reply:
Could ``SIG_ELEMENTS`` be a set?

Reply:
I need to remember why I used a list. I think I wanted to retain the order of the classes but I think it does not matter (and having a set is probably safer).

Also I did not put `#:` because I did not wanted to expose the documentation lines and instead wanted to write a corresponding .rst page (but I have no idea where to write it actually since this is fairly an advanced feature).

Reply:
> but I have no idea where to write it

``extdev/nodes.rst`` has the current documentation for ``addnodes``, that seems a good home?